cpp/src/arrow/acero/aggregate_benchmark.cc

```diff
@@ -866,5 +890,47 @@ BENCHMARK(TDigestKernelDoubleMedian)->Apply(QuantileKernelArgs);
 BENCHMARK(TDigestKernelDoubleDeciles)->Apply(QuantileKernelArgs);
 BENCHMARK(TDigestKernelDoubleCentiles)->Apply(QuantileKernelArgs);
 
+//
+// RowSegmenter
+//
+
+template <typename... Args>
+static void BenchmarkRowSegmenter(benchmark::State& state, Args&&...) {
```

Comment:
Can we turn this into a more realistic GroupBy benchmark? For example:
* M=0 to 1 non-segmented keys
* N=1 to 2 segmented keys
* K=key cardinality (2,8,64)

Reply:
Sure. Will do, and the name should be named with `GroupBy` for sure.

Reply:
Updated.